scripts/createReleasePR.mjs

```diff
@@ -198,11 +198,123 @@ async function findForkRemote() {
   }
 }
 
+/**
+ * Find the latest major version from the upstream remote
+ * @param {string} remote - The name of the remote (e.g., 'upstream')
+ * @returns {Promise<string>} The latest major version (e.g., '7', '6', etc.)
+ */
+async function findLatestMajorVersion() {
+  try {
+    // fetch tags from the GitHub API and return the last one
+    const { data: tags } = await octokit.rest.repos.listTags({
+      owner: ORG,
+      repo: REPO,
+    });
+    const tagName = tags[0].name.trim();
+    const match = tagName.match(/^v(\d+)\.(\d+)\.(\d+)/);
+    if (!match) {
+      console.error(`Error: Unable to parse version from tag ${tagName}`);
+      process.exit(1);
+    }
+    return match[1];
+  } catch (error) {
+    console.error('Error finding latest major version:', error);
+    process.exit(1);
+  }
+}
+
+/**
+ * Compares and sorts version strings extracted from tags following semantic versioning logic.
+ * @param {string} a The first tag string prefixed with 'v' (e.g., 'v1.2.3-alpha.1').
+ * @param {string} b The second tag string prefixed with 'v' (e.g., 'v1.2.3').
+ * @return {number} A negative number if `a` is less than `b`, a positive number if `a` is greater than `b`, or 0 if they are equal.
+ */
+function sortVersionsFromTags(a, b) {
+  // Sort versions using semver logic
+  // Remove 'v' prefix
+  const aVersion = a.substring(1);
+  const bVersion = b.substring(1);
+
+  // Split into version parts and prerelease parts
+  const [aVersionPart, aPrereleasePart] = aVersion.split('-');
+  const [bVersionPart, bPrereleasePart] = bVersion.split('-');
+
+  // Compare version parts (major.minor.patch)
+  const aParts = aVersionPart.split('.').map(Number);
+  const bParts = bVersionPart.split('.').map(Number);
+
+  for (let i = 0; i < 3; i += 1) {
+    if (aParts[i] !== bParts[i]) {
+      return aParts[i] - bParts[i];
+    }
+  }
+
+  // If version parts are equal, handle prerelease parts
+
+  // If one has prerelease and the other doesn't, the one with prerelease is greater
```

Comment:
When comparing "v8.0.0-alpha.0" and "v8.0.0", shouldn't "v8.0.0" be greater than "v8.0.0-alpha.0"? It seems contrary to this comment

Reply:
oh damn ... you are right ... sry